app/site/controllers.py

```diff
@@ -128,15 +128,12 @@ def publisher_dashboard(publisher):
 
 @site_blueprint.route("/search", methods=["GET"])
 def search_package():
-    try:
-        q = request.args.get('q')
-        if not q.strip():
-            return render_template("search.html", title='Search', query_term=''), 200
-
+    q = request.args.get('q').strip()
+    if q:
         datapackage_list = DataPackageQuery(query_string=q).get_data()
```

Comment:
Do you limit the number of search results. We should not show more than 20 or so results at a time.